src/lem/multiframe.rs

```diff
@@ -836,6 +836,27 @@ impl<'a, F: LurkField, C: Coprocessor<F>> nova::traits::circuit::StepCircuit<F>
     {
         assert_eq!(self.arity(), z.len());
 
+        if cs.is_witness_generator() {
```

Comment:
Does it make sense to check that `cs.aux_slice().is_empty() && cs.inputs_slice().len() == 1` as well, to ensure some degree of idempotence & sanity-checking ?